src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -0,0 +1,83 @@
+package dev.openfeature.sdk;
+
+import org.apache.logging.log4j.util.TriConsumer;
+
+/**
+ * Abstract EventProvider. Providers must extend this class to support events.
+ * Emit events with {@link #emit(ProviderEvent, ProviderEventDetails)}. Please
+ * note that the SDK will automatically emit
+ * {@link ProviderEvent#PROVIDER_READY } or
+ * {@link ProviderEvent#PROVIDER_ERROR } accordingly when
+ * {@link FeatureProvider#initialize(EvaluationContext)} completes successfully
+ * or with error, so these events need not be emitted manually during
+ * initialization.
+ *
+ * @see FeatureProvider
+ */
+public abstract class EventProvider implements FeatureProvider {
```

Comment:
This is the class _application authors_ must implement to use events. See javadoc above.